app/src/main/AndroidManifest.xml

```diff
@@ -1,6 +1,15 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.bucketlist">
+    <!--Give Internet Permission and Location Permission-->
+    <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
+    <uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
+    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
+    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
+    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
+
```

Comment:
I added internet permission because considering one of your requirements is that given a current set of GPS coordinates or city/state and a keyword list, when the user clicks the button, search for matches within a 100 miles of that city/state or coordinates, Then display matches. 
Check out: https://developers.google.com/maps/documentation/android-sdk/location
and 
https://www.youtube.com/watch?v=VgZZemAwLTk